src/sirius_cameras/src/oak_d_w.cpp

```diff
@@ -0,0 +1,287 @@
+#include <depthai_bridge/ImuConverter.hpp>
+
+#include "depthai/depthai.hpp"
+
+std::vector<std::string> usbStrings = { "UNKNOWN", "LOW", "FULL", "HIGH", "SUPER", "SUPER_PLUS" };
+
+auto getMonoResolution(std::string resolution)
+{
+  typedef dai::node::MonoCamera::Properties::SensorResolution monoResolution;
+  if (resolution == "720p")
+  {
+    return std::make_tuple(1280, 720, monoResolution::THE_720_P);
+  }
+  else if (resolution == "400p")
+  {
+    return std::make_tuple(640, 400, monoResolution::THE_400_P);
+  }
+  else if (resolution == "800p")
+  {
+    return std::make_tuple(1280, 800, monoResolution::THE_800_P);
+  }
+  else if (resolution == "480p")
+  {
+    return std::make_tuple(640, 480, monoResolution::THE_480_P);
+  }
+  else
+  {
+    ROS_ERROR("Invalid parameter. -> monoResolution: %s", resolution.c_str());
+    throw std::runtime_error("Invalid mono camera resolution.");
+  }
+}
+auto getRGBResolution(std::string resolution)
+{
+  typedef dai::node::ColorCamera::Properties::SensorResolution rgbResolution;
+  if (resolution == "1080p")
+  {
+    return std::make_tuple(1920, 1080, rgbResolution::THE_1080_P);
+  }
+  else if (resolution == "4K")
+  {
+    return std::make_tuple(3840, 2160, rgbResolution::THE_4_K);
+  }
+  else if (resolution == "12MP")
+  {
+    return std::make_tuple(4056, 3040, rgbResolution::THE_12_MP);
+  }
+  else if (resolution == "13MP")
+  {
+    return std::make_tuple(4208, 3120, rgbResolution::THE_13_MP);
+  }
+  else
+  {
+    ROS_ERROR("Invalid parameter. -> rgbResolution: %s", resolution.c_str());
+    throw std::runtime_error("Invalid color camera resolution.");
+  }
+}
+auto addMonoPipeline(dai::Pipeline& pipeline, int fps, dai::node::MonoCamera::Properties::SensorResolution resolution,
+                     std::string name, dai::CameraBoardSocket socket)
+{
+  auto monoCamera = pipeline.create<dai::node::MonoCamera>();
+  // auto xOut = pipeline.create<dai::node::XLinkOut>();
+  // xOut->setStreamName(name);
+
+  monoCamera->setResolution(resolution);
+  monoCamera->setBoardSocket(socket);
+  monoCamera->setFps(fps);
+
+  // monoCamera->out.link(xOut->input);
+  return monoCamera;
+}
+void addRGBPipeline(dai::Pipeline& pipeline, int fps, dai::node::ColorCamera::Properties::SensorResolution resolution,
+                    std::string name, dai::CameraBoardSocket socket)
+{
+  auto rgbCamera = pipeline.create<dai::node::ColorCamera>();
+  auto xOut = pipeline.create<dai::node::XLinkOut>();
+  xOut->setStreamName(name);
+
+  rgbCamera->setResolution(resolution);
+  rgbCamera->setBoardSocket(socket);
+  rgbCamera->setFps(fps);
+
+  rgbCamera->isp.link(xOut->input);
+}
+auto addDepthPipeline(dai::Pipeline& pipeline, int fps, dai::node::ColorCamera::Properties::SensorResolution resolution,
+                    std::string name, dai::CameraBoardSocket socket)
+{
+  auto stereo = pipeline.create<dai::node::StereoDepth>();
+  // auto xOut = pipeline.create<dai::node::XLinkOut>();
```

Comment:
I decide to believe all these commented-out lines are neccessary. If not I believe you should consider deleting them next time you work on this file

Reply:
I forgot to remove them. It's fixed in [05153c6](https://github.com/SKA-Robotics/rover-soft/pull/10/commits/05153c6967e977b57f51b74ed60a714cf21efd0e)